File: samples/swaggersocket-cxf-echo/src/main/java/com/wordnik/swaggersocket/samples/SwaggerSocketResource.java
/**
 *  Copyright 2015 Reverb Technologies, Inc.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package com.wordnik.swaggersocket.samples;

import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.PUT;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;

@Path("/swaggersocket")
@Produces({"text/plain"})
public class SwaggerSocketResource {

  @Path("/echo")
  @POST
  public String echo(String m) {
    // for testing a runtime exception handling
    if ("secret".equals(m)) {
        throw new RuntimeException("no secret");
    }
    return m;
  }

  @Path("/ohce")
  @POST
  public String ohce(String m) {
    // for testing a runtime exception handling
    if ("secret".equals(m)) {
        throw new RuntimeException("no secret");
    }
    return new StringBuilder(m.length()).append(m).reverse().toString();
  }

  @Path("/xbox/{word}")
  @GET
  @Produces({"application/xml"})
  public Box xbox(@PathParam("word") String word) {
    return new Box("SwaggerSocket in Action", word);
  }

  @Path("/jbox/{word}")
  @GET
  @Produces({"application/json"})
  public Box jbox(@PathParam("word") String word) {
    return new Box("SwaggerSocket in Action", word);
  }

  //for empty-entity test #27
  @Path("/put")
  @PUT
  public void put(String m) {
      System.out.println("Putting " + m);
  }
  
}

File: samples/swaggersocket-echo/src/main/scala/com/wordnik/swaggersocket/samples/SwaggerSocketResource.scala
/**
 *  Copyright 2016 SmartBear Software
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package com.wordnik.swaggersocket.samples

import javax.ws.rs.{Produces, GET, POST, PUT, Path, PathParam}

@Path("/swaggersocket")
@Produces(Array("text/plain"))
class SwaggerSocketResource {

  @Path("/echo")
  @POST
  def echo(m: String): String = {
    // for testing a runtime exception handling
    if (m == "secret") {
        throw new RuntimeException("no secret")
    } else if (m.startsWith("sleep")) {
        // sleep n
        try {
            Thread.sleep(m.substring(6).trim.toInt * 1000)
            return "Good Morning"
        } catch {
            // ignore;
            case e:Exception => {}
        }
    }
    m
  }

  @Path("/ohce")
  @POST
  def ohce(m: String): String = {
    // for testing a runtime exception handling
    if (m == "secret") {
        throw new RuntimeException("no secret")
    }
    new StringBuilder(m.length).append(m).reverse.toString
  }

  @Path("/xbox/{word}")
  @GET
  @Produces(Array("text/xml"))
  def xbox(@PathParam("word") word: String): Box = {
    new Box("SwaggerSocket in Action", word)
  }

  @Path("/jbox/{word}")
  @GET
  @Produces(Array("application/json"))
  def jbox(@PathParam("word") word: String): Box = {
    new Box("SwaggerSocket in Action", word)
  }

  //for empty-entity test #27
  @Path("/put")
  @PUT
  def put(m: String): Unit = {
      System.out.println("Putting " + m);
  }

}